radiation/FatesTwoStreamUtilsMod.F90

```diff
@@ -230,24 +251,30 @@ subroutine FatesConstructRadElements(site,fcansno_pa,coszen_pa)
             end if
 
             ! If the layer is overfull, remove some from area from
-            ! the first element that is 10x larger than the threshold
+            ! the element with the largest footprint
 
             if_overfull: if( (canopy_frac(ican)-1._r8)>area_err_thresh ) then
+
+               ! First find the element with the largest footprint
+               icolmax = -1
+               areamax = 0
                do icol = 1,n_col(ican)
-                  if(twostr%scelg(ican,icol)%area > 10._r8*(canopy_frac(ican)-1._r8))then
-                      area_ratio = (twostr%scelg(ican,icol)%area + (1._r8-canopy_frac(ican)))/twostr%scelg(ican,icol)%area
-                     twostr%scelg(ican,icol)%area = twostr%scelg(ican,icol)%area * area_ratio
-                     twostr%scelg(ican,icol)%lai  = twostr%scelg(ican,icol)%lai / area_ratio
-                     twostr%scelg(ican,icol)%sai  = twostr%scelg(ican,icol)%sai / area_ratio
-                     canopy_frac(ican) = 1.0_r8
-                     exit if_overfull
+                  if(twostr%scelg(ican,icol)%area>areamax) then
+                     icolmax = icol
+                     areamax = twostr%scelg(ican,icol)%area
                   end if
                end do
                
-               !write(fates_log(),*) 'overfull areas'
-               !twostr%cosz = coszen_pa(ifp)
-               !     call twostr%Dump(1,lat=site%lat,lon=site%lon)
-               !     call endrun(msg=errMsg(sourcefile, __LINE__))
+               ! Test out a simpler way to correct area errors
+               if(do_simple_area_correct) then
+                  twostr%scelg(ican,icolmax)%area = twostr%scelg(ican,icolmax)%area - (canopy_frac(ican)-1._r8)
```

Comment:
The `canopy_frac(ican)` here is going to be greater than `1` at this point in the code correct?  Is it possible that the value of `canopy_frac(ican)-1._r8` could be greater than `area`?  If so we might end up with a negative final area and should have a limiter against this.

Reply:
The area being used here is from the element that has the largest area of the group. The group sums to 1. In a worst case scenario, lets say we had 1000 cohorts all sharing one layer, then the area of the largest element would be no less than 0.001.  So in this case, the error would have to be larger than 0.001. Therefore, I don't think this should be a problem, but it shouldn't hurt to add in a check and complain/fail gracefully if this does happen.

Reply:
I added in a check. I put it behind a debug for now. I can run some tests to see if they trigger, and then set the debug flags back to false so that they don't slow down the run.  Note that we really should not have layers that are that overfull, this is because we try to make the cohorts fit correctly during the ppa demotion/promotion scheme. These methods here try to make the area convergence tolerance even tighter, so that it doesn't introduce energy balance errors into the model.

Reply:
30 day f45 test generated no errors on the checks , switching developer debug flag back to false